src/garage/envs/gym_env.py

```diff
@@ -0,0 +1,409 @@
+"""Wrapper class that converts gym.Env into GymEnv."""
+
+import copy
+import math
+
+import akro
+import gym
+from gym.wrappers.time_limit import TimeLimit
+
+from garage import Environment, EnvSpec, EnvStep, StepType
+
+# The gym environments using one of the packages in the following lists as
+# entry points don't close their viewer windows.
+KNOWN_GYM_NOT_CLOSE_VIEWER = [
+    # Please keep alphabetized
+    'gym.envs.atari',
+    'gym.envs.box2d',
+    'gym.envs.classic_control'
+]
+
+KNOWN_GYM_NOT_CLOSE_MJ_VIEWER = [
+    # Please keep alphabetized
+    'gym.envs.mujoco',
+    'gym.envs.robotics'
+]
+
+
+def _get_env_time_limit(env):
+    """Get time limit from a gym.Env.
+
+    Args:
+        env (gym.Env): the input gym.Env
+
+    Returns:
+        int: if there max_episode_length is found in the gym.Env. Or None if
+        not found.
+
+    Raises:
+        RuntimeError: if the gym.Env is wrapped by a gym.TimeLimit,
+        and env.spec._max_episode_steps and env._max_episode_steps don't match.
+    """
+    gym_spec_steps = None
+    if hasattr(env, 'spec') and env.spec:
+        gym_spec_steps = env.spec.max_episode_steps
+    else:
+        # metaworld env doesn't have spec, what to do?
```

Comment:
not all gym.Envs have specs. it's not necessarily part of the API -- so you need to handle it gracefully. You can still get env.observation_space and env.action_space.

Please leave an issue on metaworld to populate `spec`